### PR TITLE
Blackmagickery and hackery

### PR DESCRIPTION
**Pull request by Soveu**
Solves #2 in a _beautiful manner_ \s

### CODE REVIEW

File: lib/src/lib.rs [hunk 1]
```diff
@@ -298,11 +299,58 @@ impl<'s> quote::ToTokens for MethodTest<'s> {
...
+
+                    if f == WhichFailed::Second {
+                        panic!("Implementation panicked while the model did not");
+                    }
+
+                    if let WhichFailed::None(test_passed) = f {
+                        assert!(test_passed, "The values aren't equal");
+                    }
+
+                    /* First paniced, check if second also does */
```
Comment (Plecra) [inline lib/src/lib.rs:348]: I'm pretty sure we'll also get to this line if both calls return the same value. (`WhichFailed::None(true)`)

Comment (Soveu) [reply lib/src/lib.rs:348]: Yeah, if the assertion is true, the code continues to execute

**Comment by Soveu** [comment]
I also though about using `Cell` here instead of a guard, here compiler also shouldn't optimize out the assignment, but the code looks cleaner
[url to rust playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3a6beeb2f94a9af026f4477b2fbda071)

**Comment by Soveu** [comment]
Without `Cell` compiler throws out warnings about value being assigned, but not read

**Comment by Plecra** [comment]
What's the issue? Can you explain why https://github.com/jakubadamw/rutenspitz/pull/4#issuecomment-661405733 doesn't apply?

File: lib/src/lib.rs (hunk 1 above)
Comment (Plecra) [reply lib/src/lib.rs:348]: Right, so why do we want to check the behavior of `tested` again if we already know it's the same?

Comment (Plecra) [reply lib/src/lib.rs:348]: Right, so why do we want to check the behavior of tested again if we already know it's the same?

**Comment by Soveu** [comment]
Its because the macro does not define `model` and `tested` as mutable

**Comment by jakubadamw** [comment]
@Soveau, thanks very much for this PR, and @Plecra for the input.

I cleaned it up and addressed the issued described in [this comment](https://github.com/jakubadamw/rutenspitz/pull/6#discussion_r457732276).

**Comment by Soveu** [comment]
Closing because of #7

**Comment by jakubadamw** [comment]
Yeah, it'd close automatically once I merge that one, which I'll do once it goes green. 🙂

**Pull request closed without merging**